### PR TITLE
Add boto_elasticsearch_domain module

### PR DESCRIPTION
**Pull request by kraney**
### What does this PR do?

Adds a module for managing Amazon Elasticsearch Service domains.
### What issues does this PR fix or reference?

None
### Previous Behavior

Functionality did not exist.
### New Behavior

Adds both module and state module for elasticsearch_domain.
### Tests written?
- [x] Yes
- [ ] No

### CODE REVIEW

File: salt/modules/boto_elasticsearch_domain.py [hunk 1]
```diff
@@ -0,0 +1,469 @@
...
+        domain = conn.describe_elasticsearch_domain(DomainName=DomainName)
+        return {'exists': True}
+    except ClientError as e:
+        if e.response.get('Error', {}).get('Code') == 'ResourceNotFoundException':
+            return {'exists': False}
+        return {'error': salt.utils.boto3.get_error(e)}
+
+
+def status(DomainName,
+             region=None, key=None, keyid=None, profile=None):
```
Comment (rallytime) [inline salt/modules/boto_elasticsearch_domain.py]: Same comment here about `DomainName` to `domain_name`. Can you also line these kwargs up with the line above?
```
def status(domain_name,
                 region=None, key=None, keyid=None, profile=None)
```

File: salt/states/boto_elasticsearch_domain.py [hunk 2]
```diff
@@ -0,0 +1,359 @@
+# -*- coding: utf-8 -*-
+'''
+Manage Elasticsearch Domains
+=================
+
+.. versionadded:: 2016.3.0
```
Comment (rallytime) [inline salt/states/boto_elasticsearch_domain.py]: This should be `Carbon` instead of `2016.3.0`

**Comment by rallytime** [comment]
@kraney This addition is great, as usual! I am going to be a little picky here, though, and ask that your arg/kwarg names are changes to lower case letters using underscores instead of camelCase, as stated in our style docs: https://jenkins.saltstack.com/job/salt-pr-lint-n/14152/violations/file/salt/modules/boto_elasticsearch_domain.py/

That way we can keep this more consistent with other Salt modules.

There's also a small pylint error: https://jenkins.saltstack.com/job/salt-pr-lint-n/14152/violations/file/salt/modules/boto_elasticsearch_domain.py/

Other than that, this looks awesome!

**Comment by kraney** [comment]
We had a prior discussion around this naming convention in #29573 and #30327. The names are selected to match the values returned by boto3 when you do a 'describe'. That way, inputs are symmetric to outputs. I'm hopeful I can use a consistent policy across all boto3-based modules.

**Comment by rallytime** [comment]
@kraney Oh, actually, it appears there were some related test failures here, too. There are only two failures related to your change, they're the ones labeled `unittest.loader.ModuleImportFailure.boto_elasticsearch_domain_test`. Would you mind taking a look at those? If you run into trouble let us know and we can help out.

The other test failures (gpg and netapi.rest_tornado) do not apply to this change and we'll be fixing them elsewhere.

**Comment by rallytime** [comment]
Thanks @kraney!

**Pull request merged**